arangod/Agency/Agent.cpp

```diff
@@ -656,21 +674,15 @@ void Agent::advanceCommitIndex() {
     
   term_t t = _constituent.term();
   {
-    _tiLock.assertNotLockedByCurrentThread();
-    MUTEX_LOCKER(_ioLocker, _ioLock);
-    index_t commitIndex;
-    {
-      CONDITION_LOCKER(guard, _waitForCV);
-      commitIndex = _commitIndex;
-    }
-    if (index > commitIndex) {
+    MUTEX_LOCKER(oLocker, _oLock);
```

Comment:
previous code cached commitIndex the proceeded to applyLogEntries().  This will now hold the _oLock throughout the applyLogEntries() and use _commitIndex directly.  Is it possible that the lock will be held "too long" and have side effects?

wait, is _oLock intended protection for both _readDB & _commitIndex?  if so, comment withdrawn

Reply:
yes. cannot give up that dependancy as far i can see